extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/RouteHandlers.java

```diff
@@ -18,14 +19,22 @@ private RouteHandlers() {
     private static final LazyValue<Event<SecurityIdentity>> SECURITY_IDENTITY_EVENT = new LazyValue<>(
             RouteHandlers::createEvent);
 
-    public static void setContentType(RoutingContext context) {
+    public static void setContentType(RoutingContext context, String defaultContentType) {
         HttpServerResponse response = context.response();
-        if (response.headers().get(CONTENT_TYPE) == null) {
-            String acceptableContentType = context.getAcceptableContentType();
-            if (acceptableContentType != null) {
-                response.putHeader(CONTENT_TYPE, acceptableContentType);
+        response.headersEndHandler(new Handler<Void>() {
```

Comment:
Shouldn't we use `io.vertx.ext.web.RoutingContext.addHeadersEndHandler(Handler<Void>)` instead?

Also from perf POV this means one more `Handler` instance per each request. It's probably negligible but I wonder if the previous approach "set the content type after the route method is executed" represents some serious issue?

Reply:
The previous approach would fail if the method used the RoutingContext to send data, as the headers would have already been written.

Reply:
Ah, you're right! What about the `RoutingContext.addHeadersEndHandler()`?

Reply:
Oops, forgot to push it.